src/test/test_failure_slice.cpp

```
@@ -292,3 +292,48 @@ TEST_CASE("empty seed assertion still includes failing label", "[failure_slice][
+TEST_CASE("compute_failure_slices reports all independent failures for loop.o", "[failure_slice][integration]") {
+    const std::string sample = "ebpf-samples/build/loop.o";
+    if (!sample_exists(sample)) {
+        SKIP("Sample file not found: " << sample);
+    }
+    const auto slices = get_failure_slices(sample, "test_md");
+
+    REQUIRE(slices.size() == 2);
+    REQUIRE(slices[0].failing_label != slices[1].failing_label);
+}
+
+// Verify print_failure_slices renders a distinct section per slice, not just the first.
```

Cppcheck (2.21.0)

[style] The function 'signed_int' is never used.

(unusedFunction)